### PR TITLE
Marketplace: mobile category menu

### PR DESCRIPTION
**Pull request by lsl**
#### Changes proposed in this Pull Request

* Adds mobile only handling for the category selector to replace dropdown with a swipe / scroll selector in its place

#### Testing instructions

* On mobile breakpoints < 660px, you should see the swipe-friendly category selector and be able to change category.
* At non mobile breakpoints the new selector from [this parent branch / pr](https://github.com/Automattic/wp-calypso/pull/63592) should render in its place.

https://user-images.githubusercontent.com/811776/169184383-4efcb2ee-a31a-4e5c-a429-e1c0c1969587.mp4

Related to https://github.com/Automattic/wp-calypso/issues/62891

### CODE REVIEW

**Comment by lsl** [comment]
@Rolilink easier to view the diff without [b37b026](https://github.com/Automattic/wp-calypso/pull/63787/commits/b37b0265e4bb77e29ee33b2353e40cf3956c5c3f) (e.g. [these three commits are the main change](https://github.com/Automattic/wp-calypso/pull/63787/files/8c0dbf1375b6f6ab89b548d7708653b065072486)) as all I'm doing in that final refactor commit is moving swipe menu into the responsive toolbar group component and pushing the existing toolbar group into a subcomponent "dropdown group".

File: client/my-sites/plugins/categories/use-categories.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ export function useCategories(
 	}
 
 	const categories = {
-		discover: { name: __( 'Discover' ), slug: 'discover', tags: [] },
+		discover: { name: __( 'All' ), slug: 'discover', tags: [] },
```
Comment (lsl) [inline client/my-sites/plugins/categories/use-categories.tsx]: Matching figma

File: client/components/fixed-navigation-header/index.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import Breadcrumb, { Item as TBreadcrumbItem } from 'calypso/components/breadcru
 
 const Header = styled.header`
 	position: fixed;
-	z-index: 1;
+	z-index: 10;
```
Comment (lsl) [inline client/components/fixed-navigation-header/index.tsx:7]: Core's button components have z-index of 1, moving our header components to 10 and 20 to keep them on top while scrolling.

**Comment by lsl** [comment]
Should be good to merge at this point.

There are a few issues I want to fix but I've put them in follow up issues to solve as the size of this got pretty silly with the refactor step added in.

File: client/my-sites/plugins/categories/responsive-toolbar-group/style.scss [hunk 3]
```diff
@@ -35,27 +35,49 @@
...
-					// disable blue box on focus
-					box-shadow: none !important;
-					outline: none !important;
-				}
+	.responsive-toolbar-group__button-item, .responsive-toolbar-group__menu-item {
+		&:focus {
+			&::before {
+				// disable blue box on focus
+				box-shadow: none !important;
+				outline: none !important;
```
Comment (lsl) [inline client/my-sites/plugins/categories/responsive-toolbar-group/style.scss:51]: just tabbing in

**Review by paulopmt1 — APPROVED** [review]
Everything is working fine. Good job @lsl!

I have two visual non-blocker questions:

- The extra space between Most popular searches and mobile categories seems unnecessary if we don't use the "header divisor". Should we add that divisor line [from the design:](https://dotcommarketplacep2.wordpress.com/2022/05/09/category-filter-refinements/)?

image

- The scroll seems to be too close when we are testing with a mobile device in our hands. Should we add some padding separating it from the category a little bit? Maybe @SaxonF could help here.


image

**Comment by Rolilink** [comment]
> * The extra space between Most popular searches and mobile categories seems unnecessary if we don't use the "header divisor". Should we add that divisor line [from the design:](https://dotcommarketplacep2.wordpress.com/2022/05/09/category-filter-refinements/)?

@paulopmt1 I think the same maybe we can add a follow-up ticket to add that divisor line?. To my eyes is not actually a divisor line but a change in background color.

> * The scroll seems to be too close when we are testing with a mobile device in our hands. Should we add some padding separating it from the category a little bit? Maybe @SaxonF could help here.

I think we can fine tune this behavior a little more, maybe using this one https://stackoverflow.com/questions/20997183/how-to-hide-scrollbar-in-firefox.

**Comment by Rolilink** [comment]
LGTM too! hiding scrollbar on the last commit and merging

File: client/my-sites/plugins/categories/use-categories.tsx (hunk 1 above)
Comment (Rolilink) [inline client/my-sites/plugins/categories/use-categories.tsx]: Not sure about this change since we are not showing all the plugins, but the discover section ones I think this maybe a placeholder in design what do you think @SaxonF. Gonna revert this change

**Comment by lsl** [comment]
Thanks for merge + fixes, not sure about the scrollbar removal added #63831 to push to @SaxonF to decide. I'm not really fussed either way but did kind of like the feedback myself so wondering if its something we want or not.

**Comment by SaxonF** [comment]
I think hiding scrollbar on mobile is ok. I'll put together a separate issue with a few style tweaks such as the divider and spacing refinements.
